avocado/utils/kernel.py

```diff
@@ -64,12 +64,8 @@ def download(self):
         """
         self.kernel_file = self.SOURCE.format(version=self.version)
         full_url = self.URL + self.SOURCE.format(version=self.version)
-        path = os.path.join(self.work_dir, self.kernel_file)
-        if os.path.isfile(path):
-            log.info("File '%s' exists, will not download!", path)
-        else:
-            log.info("Downloading '%s'...", full_url)
-            download.url_download(full_url, path)
+        asset.Asset(full_url, asset_hash=None, algorithm=None,
+                    locations=None, cache_dirs=[self.work_dir]).fetch()
```

Comment:
Btw how about adding `data_dir` entry for assets to avoid custom handling (eg. inside `Test` class)?